### PR TITLE
Update tfms

### PR DESCRIPTION
**Pull request by sebastienros**
I realized the net8.0 was not correctly setting the feature flags and was not optimized :/
Then thought I could just remove netstandard and add the specific ones since when using netstandard you still need to add an core one (recommendation).

I will definitely change my mind, but for now this is what it is.

### CODE REVIEW

File: src/Parlot/Parlot.csproj [hunk 1]
```diff
@@ -14,16 +14,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="System.Memory" Condition="'$(TargetFramework)' == 'netstandard2.0'" />
-    <PackageReference Include="System.Collections.Immutable" Condition="'$(TargetFramework)' == 'netstandard2.0' or '$(TargetFramework)' == 'netstandard2.1'" />
+    <PackageReference Include="System.Memory" Condition="'$(TargetFramework)' == 'net462' "/>
```
Comment (lahma) [inline src/Parlot/Parlot.csproj:17]:
```suggestion
    <PackageReference Include="System.Memory" Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))" />
```

File: src/Parlot/Parlot.csproj [hunk 2]
```diff
@@ -14,16 +14,9 @@
...
+    <PackageReference Include="System.Collections.Immutable" Condition="'$(TargetFramework)' == 'net462' or '$(TargetFramework)' == 'net6.0'" />
```
Comment (lahma) [inline src/Parlot/Parlot.csproj:18]:
```suggestion
    <PackageReference Include="System.Collections.Immutable" Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))" />
```

File: src/Parlot/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -1,3 +1,3 @@
-#if !NET8_0_OR_GREATER
```
Comment (lahma) [inline src/Parlot/Properties/AssemblyInfo.cs:1]: This should be always "on" with pre-compiler directives.

should add to csproj:

```
<PackageReference Include="PolySharp" Version="1.14.1" PrivateAssets="all" />
```
then remove files `IsExternalInit.cs` and `SkiptLocalsInitAttribute.cs`

File: src/Parlot/Parlot.csproj [hunk 4]
```diff
@@ -14,16 +14,9 @@
...
     <PackageReference Include="FastExpressionCompiler.Internal.src" PrivateAssets="all" />
```
Comment (lahma) [inline src/Parlot/Parlot.csproj:19]:
```suggestion
    <PackageReference Include="FastExpressionCompiler.Internal.src" PrivateAssets="all" />
    <PackageReference Include="PolySharp" Version="1.14.1" PrivateAssets="all" />
```

**Comment by sebastienros** [comment]
Thanks for the feedback Marko

**Pull request merged**